Assets/Scripts/Entity/Player.cs

```diff
@@ -45,11 +48,18 @@ public void Hurt(Vector2 KnockBack)
         {
             Health--;
             if (Health < 1) {
-                transform.position = _spawnPoint;
-                Health = _maxHealth;
+                Kill();
             } else 
                 RB.AddForce(KnockBack);
-        }  
+        }
+
+        [Command("Kill")]
+        public void Kill()
+        {
+            EventSystem.Current.PlayerDeath(this);
```

Comment:
Object reference not set to an instance of an object.

Reply:
Problem related to Eventsystem.

Reply:
This is cause EventSystem does not exist in that scene